relay-server/src/envelope.rs

```diff
@@ -959,19 +958,9 @@ impl Envelope {
 
     /// Return the data category type of the event item, if any, in this envelope.
     pub fn get_event_category(&self) -> Option<DataCategory> {
-        let event_categories: Vec<DataCategory> = self
-            .items()
-            .filter_map(infer_event_category)
-            .unique()
-            .collect();
-        if event_categories.len() == 1 {
-            Some(event_categories[0])
-        } else {
-            if event_categories.len() > 1 {
-                relay_log::warn!("Conflicting event categories in the same envelope");
-            }
-            None
-        }
+        self.items().find_map(infer_event_category)
+        // There are some cases where multiple items may have different categories, but returning
+        // the first is good enough for now.
```

Comment:
Agreed 👍  

The case we will have to look at is if someone is sending a `__sentry-event` msgpack attachment + a minidump. Depending on the order, we might get `::Default` or `::Error`. For stats, they are all equal, so it's not a big problem. 

`EnvelopeLimiter` already takes care of that, so it shouldn't be too hard to abstract that out in a follow-up.